fossunited/fossunited/user_utils.py

```diff
@@ -1,13 +1,18 @@
+import re
+
 import frappe
 
 from fossunited.doctype_ids import USER_PROFILE
 
 
 def set_unique_username(doc, method):
-    full_name = doc.full_name
+    full_name = doc.full_name.lower()
     doc.first_name = full_name.split(" ")[0]
     doc.last_name = " ".join(full_name.split(" ")[1:])
-    doc.username = generate_username(doc.full_name.replace(" ", "_"))
+    initial_username = re.sub(
+        r"[^a-z0-9_]", "", full_name.replace(" ", "_")
+    )
```

Comment:
what does this do

Reply:
Just creates an initial username to pass to generate_username method - with an add-on check again for the username to be with lowercase letters, numbers and underscores. 

In case someone puts symbols in Full Name while entering it.